pandas/core/nanops.py

```diff
@@ -245,8 +245,7 @@ def _maybe_get_mask(
     """
     if mask is None:
         if is_bool_dtype(values.dtype) or is_integer_dtype(values.dtype):
-            # Boolean data cannot contain nulls, so signal via mask being None
-            return None
+            return np.broadcast_to(False, values.shape)
```

Comment:
@jbrockmendel didn't check the rest of the PR, but was this change a crucial part of the fix? 

There are bunch of benchmarks showing a slowdown (almost all Ops benchmarks involving integer and boolean data), so that might be caused by this (didn't check to be sure).

Reply:
> but was this change a crucial part of the fix?

yes

Reply:
The reason this seemed necessary was because `_maybe_null_out` was returning a wrong shape when the mask was None -> https://github.com/pandas-dev/pandas/pull/41920